src/Cli/dotnet/Commands/Solution/Add/SolutionAddCommand.cs

```diff
@@ -202,6 +204,21 @@ private void AddProject(SolutionModel solution, string fullProjectPath, ISolutio
             project.AddProjectConfigurationRule(new ConfigurationRule(BuildDimension.BuildType, solutionBuildType, "*", projectBuildType));
         }
 
+        // Get referencedprojects from the project instance
+        var referencedProjectsFullPaths = projectInstance.EvaluatedItemElements
```

Comment:
I'm wondering if there's a way you can reuse this computation between AddProject calls...probably not since each project has its own set of ProjectReferences.

Reply:
I think maybe not, because it has to open the project instance to look for this

Reply:
Yeah, unfortunately so